tycho-core/src/main/java/org/eclipse/tycho/core/maven/MavenDependencyInjector.java

```diff
@@ -127,7 +133,24 @@ private boolean requiresExternalDependencies() {
     }
 
     private Dependency createSystemScopeDependency(ArtifactKey artifactKey, File location) {
-        return createSystemScopeDependency(artifactKey, P2_GROUPID_PREFIX + artifactKey.getType(), location);
+        if (session != null && location.getAbsolutePath().startsWith(session.getLocalRepository().getBasedir())) {
+            String suffix = location.getAbsolutePath().substring(session.getLocalRepository().getBasedir().length());
+            List<String> segments = new ArrayList<>(Arrays.asList(suffix.split(File.separator)));
+            segments.removeIf(String::isEmpty);
```

Comment:
Won't something like 'Path' be better suited to examine segment instead of string splitting? or even file.getParent().getName() ...

Reply:
Yes, I'll try to rework this.